packages/astro/src/internal/__astro_component.ts

```diff
@@ -49,8 +49,41 @@ function isCustomElementTag(name: string | Function) {
 
 const rendererCache = new Map<any, RendererInstance>();
 
+/** For client:only components, attempt to infer the required renderer. */
+function inferClientRenderer(metadata: Partial<AstroComponentMetadata>) {
+  // If there's only one renderer, assume it's the required renderer
+  if (rendererInstances.length === 1) {
+    return rendererInstances[0];
+  } else if (metadata.value) {
+    // Attempt to find the renderer by matching the hydration value
+    const hint = metadata.value;
+    const match = rendererInstances
+      .find((instance) => {
+        if (!instance.source) {
+          return false;
+        }
+
+        return instance.source.indexOf(hint) >= 0;
```

Comment:
I think the is the part you were talking about. I think `source` here is a specifier that is imported maybe? I think a better thing to use is the `name` which each renderer has. We might not be passing that into this interface, and if not then `config_manager.ts` is the thing that creates these objects and name should be easy to add.

Then the algorithm I would suggest is:

1. If the name is an exact match of the hint then use that, ie: `<Component client:only="@astrojs/renderer-preact" />`
2. Allow a shorthand for the Astro builtin renderers, so basically look for a name matching `"@astrojs/renderer-" + hint` in order to match `<Component client:only="preact" />`.

I think this mirrors how Babel matches its plugin names, so there's precedent here.

Interested in how @FredKSchott and @natemoo-re feel about this.

Reply:
Yep that sounds like a good approach to me.  Shouldn't be a problem comparing against names from the config, and I like the special first-party support for `@astrojs/renderer-___` names

Reply:
I'm +1 on this approach!